analyzers/src/SonarAnalyzer.CSharp/Rules/NotAssignedPrivateMember.cs

```diff
@@ -42,42 +42,37 @@ public sealed class NotAssignedPrivateMember : SonarDiagnosticAnalyzer
+        private const string DiagnosticId = "S3459";
         private const string MessageFormat = "Remove unassigned {0} '{1}', or set its value.";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly Accessibility MaxAccessibility = Accessibility.Private;
 
-        private static readonly Accessibility maxAccessibility = Accessibility.Private;
-
-        protected override void Initialize(SonarAnalysisContext context)
+        private static readonly ISet<SyntaxKind> PreOrPostfixOpSyntaxKinds = new HashSet<SyntaxKind>
         {
+            SyntaxKind.PostDecrementExpression,
+            SyntaxKind.PostIncrementExpression,
+            SyntaxKind.PreDecrementExpression,
+            SyntaxKind.PreIncrementExpression,
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSymbolAction(
                 c =>
                 {
                     var namedType = (INamedTypeSymbol)c.Symbol;
-                    if (!namedType.IsClassOrStruct() ||
-                        HasStructLayoutAttribute(namedType) ||
-                        namedType.ContainingType != null)
+                    if (TypeDefinitionShouldBeSkipped(namedType))
                     {
                         return;
                     }
 
                     var removableDeclarationCollector = new CSharpRemovableDeclarationCollector(namedType, c.Compilation);
 
-                    var candidateFields = removableDeclarationCollector.GetRemovableFieldLikeDeclarations(
-                        new HashSet<SyntaxKind> { SyntaxKind.FieldDeclaration }, maxAccessibility)
-                        .Where(tuple => !IsInitializedOrFixed((VariableDeclaratorSyntax)tuple.SyntaxNode) &&
-                                        !HasStructLayoutAttribute(tuple.Symbol.ContainingType));
-
-                    var candidateProperties = removableDeclarationCollector.GetRemovableDeclarations(
-                        new HashSet<SyntaxKind> { SyntaxKind.PropertyDeclaration }, maxAccessibility)
-                        .Where(tuple => IsAutoPropertyWithNoInitializer((PropertyDeclarationSyntax)tuple.SyntaxNode) &&
-                                        !HasStructLayoutAttribute(tuple.Symbol.ContainingType));
-
-                    var allCandidateMembers = candidateFields.Concat(candidateProperties).ToList();
+                    var allCandidateMembers = RetrieveFielAndPropertyDeclarations(removableDeclarationCollector);
```

Comment:
typo: "field"
I would rename the method to `GetFieldAndPropertyDeclatations` or `GetCandidateDeclarations`.